### PR TITLE
Animate header

### PR DESCRIPTION
**Pull request by huwshimi**
Fade in the inspector header when navigating between components.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit `http://0.0.0.0:6543/:flags:/react/?deploy-target=bundle%2Fdjango-2`
- click on a service
- click on units
- click the header to go back
- the header should fade in between each navigation change

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3159/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3160/
Test PASSed.

File: jujugui/static/gui/src/app/components/inspector-header/inspector-header.js [hunk 1]
```diff
@@ -52,6 +52,24 @@ YUI.add('inspector-header', function() {
...
+      @param {Object} prevProps The props which were sent to the component.
+      @param {Object} prevState The state that was sent to the component.
+    */
+    componentDidUpdate: function(prevProps, prevState) {
+      // Only animate when switching between components.
+      if (this.props.activeComponent !== prevProps.activeComponent) {
+        var node = React.findDOMNode(this);
+        node.classList.remove('fade-in');
+        // Animate the header change.
+        window.requestAnimationFrame(function() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-header/inspector-header.js]: nice

**Comment by hatched** [comment]
:+1: QA :ok:

I find the 0.1s to be too fast, I preferred 0.3s animation, but I'll leave this up to you.

**Comment by anthonydillon** [comment]
Code and QA :+1: Although I do agree it is a little quick. Might be worth dropping to .3s but your call.

**Comment by huwshimi** [comment]
Thanks @hatched and @anthonydillon :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3171/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**